src/integrationTest/java/com/mongodb/hibernate/EmbeddableIntegrationTests.java

```diff
@@ -0,0 +1,272 @@
+    private static MongoCollection<BsonDocument> mongoCollection;
+
+    private SessionFactoryScope sessionFactoryScope;
+
+    @Test
+    void flattenedValues() {
+        var item = new ItemWithFlattenedValues();
+        item.id = new EmbeddableValue();
+        item.id.a = 1;
+        item.flattened1 = new EmbeddableValue();
+        item.flattened1.a = 2;
+        item.flattened2 = new EmbeddablePairValue1(3, new EmbeddablePairValue2(4, 5));
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollection.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id.a))
+                        .append("flattened1_a", new BsonInt32(2))
+                        .append("flattened2_a", new BsonInt32(3))
+                        .append("flattened2_flattened_a", new BsonInt32(4))
+                        .append("flattened2_flattened_b", new BsonInt32(5)));
+        var loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.get(ItemWithFlattenedValues.class, item.id));
+        assertEquals(item, loadedItem);
+        var updatedItem = sessionFactoryScope.fromTransaction(session -> {
+            var result = session.get(ItemWithFlattenedValues.class, item.id);
+            result.flattened1.a = -result.flattened1.a;
+            return result;
+        });
+        loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.get(ItemWithFlattenedValues.class, item.id));
+        assertEquals(updatedItem, loadedItem);
+    }
+
+    @Test
+    void nestedValues() {
+        var item = new ItemWithNestedValues();
+        item.id = new StructAggregateEmbeddableValue();
+        item.id.a = 1;
+        item.nested1 = new StructAggregateEmbeddableValue();
+        item.nested1.a = 2;
+        item.nested2 = new StructAggregateEmbeddablePairValue1(3, new StructAggregateEmbeddablePairValue2(4, 5));
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollection.find())
+                .containsExactly(new BsonDocument()
+                        // Hibernate ORM flattens `item.id` despite it being of an aggregate type
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id.a))
+                        .append("nested1", new BsonDocument("a", new BsonInt32(2)))
+                        .append(
+                                "nested2",
+                                new BsonDocument()
+                                        .append("a", new BsonInt32(3))
+                                        .append(
+                                                "nested",
+                                                new BsonDocument()
+                                                        .append("a", new BsonInt32(4))
+                                                        .append("b", new BsonInt32(5)))));
```

Comment:
Is it possible to cleanly parse multiline json strings for tests?

Reply:
To work in all situations, strings must use the canonical extended JSON format (for example, to represent `ObjectId`, or `int64`), and that format is neither particularly writeable, nor particularly readable. I am prefer writing and reading code that uses `BsonDocument` API instead.

Reply:
I did not mean it won't work for these specific tests, I meant the approach can't work generally speaking. This approach is also inconvenient in situations when the primary key is generated.  Which is why regardless of how tests are written in `BasicCrudIntegrationTests`, we do have tests written with `BsonDocument` API.

Done in f4e5391bbab07e2972836c09e548d07dd7d79774.